zebra-rpc/src/methods/get_block_template_rpcs/types/get_block_template.rs

```diff
@@ -9,6 +9,40 @@ use crate::methods::{
     GetBlockHash,
 };
 
+#[derive(Debug, serde::Deserialize)]
+#[serde(rename_all = "lowercase")]
+pub enum GetBlockTemplateRequestMode {
+    Template,
+    Proposal,
+}
```

Comment:
Please document where the list of modes comes from, and what each mode means:
https://en.bitcoin.it/wiki/BIP_0022#Block_Template_Request
https://en.bitcoin.it/wiki/BIP_0023#Block_Proposal

This will fail if a client specifies a mode we don't know about.
Would it make sense to have an `UnknownMode(String)` variant?
Or should we error, because we can't provide the correct response to an unknown mode?

Reply:
I think it should be an error.